### PR TITLE
Feature: banner button options

### LINKED ISSUE #5086: As a site owner, I can adjust the button style on the banner block
# Story

As a site owner, I can adjust the button style on the banner block

# Proposed solution

In `uids`:
  - Use https://github.com/uiowa/uids/pull/681 as a starting point. 
  - Test out the button classes in the `banner_link_classes` in UIDS. 

In `uiowa`:

- [x] Add a new "Button style" layout builder style grouping `/admin/config/content/layout_builder_style/group` to the banner block under "Display options" that offer these choices:
  - Primary Gold (default) `bttn bttn--primary bttn--caps`
  - Secondary (Black) `bttn bttn--secondary bttn--caps`
  - Outline `bttn bttn--outline bttn--tertiary bttn--caps`
  - Sans-serif outline `bttn bttn--outline bttn--tertiary bttn--sans-serif`
- [x] Update hook to set `Button: primary (default)`as the selected button style.

### PR DESCRIPTION
**Pull request by bspeare**
Resolves https://github.com/uiowa/uiowa/issues/5086

Todo:
- [x] add "button font" group with a "Thin" option to set alternate button style
- [x] update button styles to just read as "gold" , "black", "white", and "outline"
- [x] Review and merge https://github.com/uiowa/uids/pull/681 and attach as a UIDS release to this pull request
- [x] Update hook for banners setting default button style to `primary`
- [x] ~Add these button options to CTAs since these button style updates from UIDS will effect these buttons: https://github.com/uiowa/uiowa/issues/5149~ Added temporary fix: https://github.com/uiowa/uiowa/pull/5146/commits/128b884d65eaf46e3ddbe48bb4f74b0cf835c034

# How to test

- `blt frontend`
- `ddev blt ds --site=sandbox.uiowa.edu`
- Test on a number of banner examples:
- - https://sandbox.prod.drupal.uiowa.edu/banners-imagesvideos-w-black-and-white-gradients
- - https://sandbox.prod.drupal.uiowa.edu/banner-backgrounds-w-gradient-dark
- - https://sandbox.prod.drupal.uiowa.edu/banner-backgrounds-w-gradient-light
- - https://sandbox.prod.drupal.uiowa.edu/banners-and-buttons
- Verify that update hook set existing banners with gold button style
- Verify that newly created banners are set with the gold button style
- Verify that you can adjust and change the button styles on a banner using one link and multiple links. 
- Verify that you can use one of the outline styles against the dark gradient and that the outline turns white and the text turns white.  Test this option using the banner stacked style as well. 
- test all button color options with the button font "thin" option.

### CODE REVIEW

**Comment by bspeare** [comment]
I worked on this yesterday a bit.  I think all that is left for the banner buttons is the update hook described above.  I do wonder if we should increase the scope of this work to include https://github.com/uiowa/uiowa/issues/5149 based on the reasons provided in that issue.

**Comment by cory-skeers** [comment]
Taking a look at the update hook for this.

**Comment by cory-skeers** [comment]
@bspeare With the temporary workaround for CTAs, is this ready for review (along with the UIDS updates), or are you still working on updates (while "not working" 😜).

**Comment by bspeare** [comment]
Yes, I think it is ready if you want to mark it ready to review.

**Comment by joewhitsitt** [comment]
Pulling down to test

**Comment by joewhitsitt** [comment]
Sorry to nitpick but is it possible to rearrange these a bit? Headline stuff is split up, buttons seem thrown in the middle and for a moment I wondered if I could change the button height.

Screen Shot 2022-04-25 at 9 58 19 AM

**Comment by bspeare** [comment]
@joewhitsitt great point! I will work on that.

File: docroot/profiles/custom/sitenow/sitenow.install [hunk 1]
```diff
@@ -1893,3 +1893,34 @@ function sitenow_update_9039() {
 function sitenow_update_9040() {
   drupal_flush_all_caches();
 }
+
+/**
+ * Set existing banner blocks with a default button style value.
+ */
+function sitenow_update_9041() {
+  // Loop through our newly updated blocks, banner and CTAs.
```
Comment (joewhitsitt) [inline docroot/profiles/custom/sitenow/sitenow.install]: I know CTAs are on the horizon but right now this comment is misleading.
>// Loop through our newly updated blocks, banner and CTAs.

**Review by joewhitsitt — APPROVED** [review]
Except for https://github.com/uiowa/uiowa/pull/5146#issuecomment-1108690155, this seems to be working as expected. I tried several but not _all_ combinations. Small requested tweaks to lb_style order and code comments. Could merge without.

**Comment by bspeare** [comment]
Thanks @joewhitsitt! Converting to draft to attach UIDS release and make code updates.

**Comment by bspeare** [comment]
@joewhitsitt I updated the order to push the buttons towards the bottom of the options:

Screen Shot 2022-04-25 at 10 55 33 AM

**Comment by bspeare** [comment]
I've attached the UIDS release to this and I think I'll leave this in draft until after the mid-sprint review tomorrow to see if there is any further feedback.

**Comment by bspeare** [comment]
I think we can merge this in.  Since Joe's review, I added a button color contrast fix and a headline size fix for the pre-title when it is used in 3 and 4 column layouts.

**Pull request merged**